src/weights.jl

```diff
@@ -18,12 +18,25 @@ macro weights(name)
 end
 
 length(wv::AbstractWeights) = length(wv.values)
-values(wv::AbstractWeights) = wv.values
 sum(wv::AbstractWeights) = wv.sum
 isempty(wv::AbstractWeights) = isempty(wv.values)
 size(wv::AbstractWeights) = size(wv.values)
 
-Base.getindex(wv::AbstractWeights, i) = getindex(wv.values, i)
+Base.convert(::Type{AbstractVector}, wv::AbstractWeights) = wv.values
+Base.convert(::Type{Vector}, wv::AbstractWeights) = convert(Vector, wv.values)
+
+@propagate_inbounds function Base.getindex(wv::W, i::Integer) where W <: AbstractWeights
+    @boundscheck checkbounds(wv, i)
+    wv.values[i]
+end
+
+@propagate_inbounds function Base.getindex(wv::W, i::AbstractArray{<:Int}) where W <: AbstractWeights
```

Comment:
```suggestion
@propagate_inbounds function Base.getindex(wv::W, i::AbstractArray) where W <: AbstractWeights
```
(We need to accept things like `Any[1, 2]`.)